cli/internal/core/engine_test.go

```diff
@@ -107,29 +121,42 @@ func TestDependenciesOnUnspecifiedPackages(t *testing.T) {
 	//       app2 <
 	//              \ libC
 	//
-	graph := &dag.AcyclicGraph{}
-	graph.Add("app1")
-	graph.Add("app2")
-	graph.Add("libA")
-	graph.Add("libB")
-	graph.Add("libC")
-	graph.Add("libD")
-	graph.Connect(dag.BasicEdge("libA", "libB"))
-	graph.Connect(dag.BasicEdge("libB", "libD"))
-	graph.Connect(dag.BasicEdge("app0", "libA"))
-	graph.Connect(dag.BasicEdge("app1", "libA"))
-	graph.Connect(dag.BasicEdge("app2", "libB"))
-	graph.Connect(dag.BasicEdge("app2", "libC"))
-
-	p := NewEngine(graph)
-	p.AddTask(&Task{
-		Name:           "build",
-		TaskDefinition: fs.TaskDefinition{TopologicalDependencies: []string{"build"}},
-	})
-	p.AddTask(&Task{
-		Name:           "test",
-		TaskDefinition: fs.TaskDefinition{TopologicalDependencies: []string{"build"}},
+	workspaceGraph := dag.AcyclicGraph{}
+	workspaceGraph.Add("app1")
+	workspaceGraph.Add("app2")
+	workspaceGraph.Add("libA")
+	workspaceGraph.Add("libB")
+	workspaceGraph.Add("libC")
+	workspaceGraph.Add("libD")
+	workspaceGraph.Connect(dag.BasicEdge("libA", "libB"))
+	workspaceGraph.Connect(dag.BasicEdge("libB", "libD"))
+	workspaceGraph.Connect(dag.BasicEdge("app0", "libA"))
+	workspaceGraph.Connect(dag.BasicEdge("app1", "libA"))
+	workspaceGraph.Connect(dag.BasicEdge("app2", "libB"))
+	workspaceGraph.Connect(dag.BasicEdge("app2", "libC"))
+
+	buildTask := fs.TaskDefinition{TopologicalDependencies: []string{"build"}}
+	testTask := fs.TaskDefinition{TopologicalDependencies: []string{"build"}}
+
+	pipeline := fs.Pipeline{"build": buildTask, "test": testTask}
+
+	p := NewEngine(&graph.CompleteGraph{
+		WorkspaceGraph:  workspaceGraph,
+		Pipeline:        pipeline,
+		TaskDefinitions: map[string]*fs.TaskDefinition{},
+		WorkspaceInfos: graph.WorkspaceInfos{
+			"app1": &fs.PackageJSON{},
+			"app2": &fs.PackageJSON{},
+			"libA": &fs.PackageJSON{},
+			"libB": &fs.PackageJSON{},
+			"libC": &fs.PackageJSON{},
+			"libD": &fs.PackageJSON{},
+		},
```

Comment:
All these engine tests are really coupled to the implementation, so they have to change when things change. I would like to convert a lot of them to integration tests with prysk, so we can validate what we really need to validate.